tests/taint_maturity/c/taint_if.yaml

```diff
@@ -0,0 +1,17 @@
+rules:
+  - id: taint-maturity
+    mode: taint
+    languages:
```

Comment:
potentially we could have a taint_maturity/POLYGLOT/ directory to put reusable rules that can work for multiple
languages (like we have for tests/patterns/POLYGLOT/